src/worker/core/GameSim.basketball/index.ts

```diff
@@ -2338,87 +2287,83 @@ class GameSim extends GameSimBase {
 
 		let pAst;
 		let pidAst;
+		let pidDefense;
 		if (passer !== undefined) {
 			pAst = this.playersOnCourt[this.o][passer];
 			pidAst = this.team[this.o].player[pAst].id;
 		}
-
+		// Randomly pick a name to be dunked on. Used in atRim
+		if (pidFoul !== undefined) {
+			pidDefense = pidFoul;
+		} else {
+			const ratios = this.ratingArray("blocking", this.d, 5);
+			const p = this.playersOnCourt[this.d][pickPlayer(ratios)];
+			pidDefense = this.team[this.d].player[p].id;
+		}
+		let fgMakeLogType:
+			| FgMakeType
+			| "fgPutBackAndOne"
+			| "fgPutBack"
+			| "fgAtRimAndOne"
+			| "fgAtRim";
```

Comment:
I think this would be cleaner if those 4 extra ones were in `FgMakeType` and then you wouldn't have to manually type them out here (and not make a typo).

Could be like:

```
type FgMakeNormal = (what you have in FgMakeType now)
type FgMakeWithoutAst = "fgPutBackAndOne" | "fgPutBack";
type FgMakeWithDefender = "fgAtRimAndOne" | "fgAtRim";
type FgMakeType = FgMakeNormalType | FgMakeWithoutAst | FgMakeWithDefender;
```

And then use those when defining `PlayByPlayEventInputScore`.

That'd be just a little clearer and less error prone, I think.